.flake8

```diff
@@ -0,0 +1,32 @@
+#
+# The project did not use flake8 since inception so there are a number
+# of time-consuming flake8-identified improvements that are just a lot
+# of busy work.  Each of these should be disabled and code cleaned up.
+#
+# W503: Line break occurred before a binary operator
+# W504: Line break occurred after a binary operator
+# E126: Continuation line over-indented for hanging indent
+# E127: Continuation line over-indented for visual indent
+# E128: Continuation line under-indented for visual indent
+# E201: Whitespace after '('
+# E202: Whitespace before ')'
+# E203: Whitespace before ':'
+# E221: Multiple spaces before operator
+# E225: Missing whitespace around operator
+# E226: Missing whitespace around arithmetic operator
+# E231: Missing whitespace after ',', ';', or ':'
+# E261: At least two spaces before inline comment
+# E265: Block comment should start with '# '
+# E302: Expected 2 blank lines, found 0
+# E501: Line too long (xx > yy characters)
+# E502: The backslash is redundant between brackets
+# F401: Module imported but unused
+# F403: 'from module import *' used; unable to detect undefined names
+# F405: Name may be undefined, or defined from star imports: module
+#
+
+[flake8]
+ignore = W503,W504,E126,E127,E128,E201,E202,E203,E221,E225,E226,E231,E261,E265,E302,E501,E502,F401,F403,F405
```

Comment:
It's a good idea to ignore the rules which will cause the most amount of (trivial) changes. For the non-ignored ones, I'm assuming we are not compliant with everything; this means the first subsequent commit would have a bunch of style changes that fall under the non-ignored rules? Why not introduce them in this commit then?

Reply:
I didn't want to do a major change - by making it PEP8 compliant, as I felt it could cause some unintended consequences. Right now, I have put `.flake8` into the repo, which we can use in our day-to-day IDE and fix local changes, without having too much impact. We are not actively checking for PEP8 issues in travis either, as correcting PEP8 issues brings in more issues than the gain we get.

Example:
PEP8 compains about `WALinuxAgent/azurelinuxagent/common/version.py:138:55: W605 invalid escape sequence '\d'` for the following piece of code.
```
EXT_HANDLER_PATTERN = b".*/WALinuxAgent-(\d+.\d+.\d+[.\d+]*).*-run-exthandlers"
```
This, when corrected by AutoPEP, results in (note the r in front of the string)
```
EXT_HANDLER_PATTERN = rb".*/WALinuxAgent-(\d+.\d+.\d+[.\d+]*).*-run-exthandlers"
```

The catch - 2.7 and below dont support the `rb` configuration of string.

Also

>  For the non-ignored ones, I'm assuming we are not compliant with everything; this means the first subsequent commit would have a bunch of style changes that fall under the non-ignored rules?
Travis won't run the flak8 actively for now, as PEP8 compliance is a bigger hassle for us now. And we won't see complaints as of now.

My [reference](https://www.youtube.com/watch?v=wf-BqAjZb8M) for not doing too much PEP8ing. :)

Reply:
The primary goal of the PR was to beef up our Python coverage, as well as introduce codecoverage tools into our codebase. We can discuss this offline as well.